static/src/js/components/Map/Map.jsx

```diff
@@ -450,9 +554,30 @@ const Map = ({
 
     // Clear the existing granule backgrounds
     granuleBackgroundsSource.clear()
+    // Clear any existing granule highlights
+    unhighlightFeature(granuleHighlightsSource)
+
+    // Clear any existing focused granules
+    clearFocusedGranuleSource(mapRef.current)
 
     // Draw the granule backgrounds
     drawGranuleBackgrounds(granules, granuleBackgroundsSource, projectionCode)
+
+    // If there is a focused granule draw it
+    if (focusedGranuleId) {
+      drawFocusedGranule({
+        collectionId: focusedCollectionId,
+        focusedGranuleSource,
+        granuleBackgroundsSource,
+        granuleId: focusedGranuleId,
+        isProjectPage,
+        map: mapRef.current,
+        onChangeFocusedGranule,
+        onExcludeGranule,
+        shouldMoveMap: false,
+        timesIconSvg
+      })
+    }
   }, [granules, granulesKey, projectionCode])
 
   // Draw the granule outlines
```

Comment:
Does the `POSTRENDER` need a cleanup function in the use effect and should it also live inside the use effect for `[granules, granulesKey, projectionCode]`

Reply:
I'm playing around with this on my next branch, but I don't think so